src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.cs

```diff
@@ -0,0 +1,782 @@
+	{
+		HorizontalScrollValue = -1 * HorizontalOffset;
+		await RaiseRenderedContentUpdated();
+	}
+
+	private void IsActiveChanged(DependencyObject sender, DependencyProperty dp)
+	{
+		if (ResetWhenNotActive && !IsActive)
+		{
+			RemoveOffset();
+			ResetZoom();
+		}
+		if (_scrollH is not null)
+		{
+			_scrollH.Visibility = IsActive ? Visibility.Visible : Visibility.Collapsed;
+		}
+		if (_scrollV is not null)
+		{
+			_scrollV.Visibility = IsActive ? Visibility.Visible : Visibility.Collapsed;
+		}
+	}
+
+	private void UpdateScrollLimits()
+	{
+		if (_presenter?.Content is FrameworkElement fe)
+		{
+			// If Actual Width/Height * Zoom < ViewPort Width/Height there should be no scroll
+			// If Actual Width/Height * Zoom > ViewPort Width/Height the amount of scroll should
+			// be based on the difference between the two values
+			var verticalScroll = Math.Max(0, (fe.ActualHeight * ZoomLevel) - ViewPortHeight);
+			var horizontalScroll = Math.Max(0, (fe.ActualWidth * ZoomLevel) - ViewPortWidth);
+
+			HorizontalMaxScroll = horizontalScroll / 2;
+			VerticalMaxScroll = verticalScroll / 2;
+
+			HorizontalMinScroll = -1 * HorizontalMaxScroll;
+			VerticalMinScroll = -1 * VerticalMaxScroll;
+		}
+	}
+
+	private void CoerceZoomLevel(DependencyObject sender, DependencyProperty dp)
+	{
+		var control = (ZoomContentControl)sender;
+		control.ZoomLevel = Math.Clamp(control.ZoomLevel, control.MinZoomLevel, control.MaxZoomLevel);
+	}
+
+	public ZoomContentControl()
+	{
+		DefaultStyleKey = typeof(ZoomContentControl);
+
+		RegisterPropertyHandlers();
+
+		this.Loaded += OnLoaded;
+		this.SizeChanged += OnSizeChanged;
+	}
+
+	private void OnLoaded(object sender, RoutedEventArgs e)
+	{
+		Centralize();
+	}
+
+	private void OnSizeChanged(object sender, SizeChangedEventArgs args)
+	{
+		if (IsLoaded && AutoZoomToCanvasOnSizeChanged)
+		{
+			ZoomToCanvas();
+		}
+	}
+
+	protected override void OnApplyTemplate()
+	{
+		T FindTemplatePart<T>(string name) where T : class =>
+			(GetTemplateChild(name) ?? throw new Exception($"Expected template part not found: {name}"))
+			as T ?? throw new Exception($"Expected template part '{name}' to be of type: {typeof(T)}");
+		_grid = FindTemplatePart<Grid>(TemplateParts.RootGrid);
+		_presenter = FindTemplatePart<ContentPresenter>(TemplateParts.Presenter);
+		_scrollV = FindTemplatePart<ScrollBar>(TemplateParts.VerticalScrollBar);
+		_scrollH = FindTemplatePart<ScrollBar>(TemplateParts.HorizontalScrollBar);
+
+		ResetOffset();
+		ResetZoom();
+
+		RegisterToControlEvents();
+		RegisterPointerHandlers();
+	}
+
+	#region ScrollBars Events
+
+	private void RegisterToControlEvents()
+	{
+		if (_presenter?.Content is FrameworkElement { } fe)
+		{
+			fe.LayoutUpdated += (s, e) =>
+			{
+				ViewPortWidth = fe.ActualWidth;
+				ViewPortHeight = fe.ActualHeight;
+
+				UpdateScrollLimits();
+			};
+		}
+
+		//due to templatebinding there's no TwoWay mode. We need to manually update the values
```

Comment:
you can:
```
<Example PropertyA="{Binding Asd, RelativeSource={RelativeSource TemplatedParent}, Mode=TwoWay}" />
```

Reply:
but look at the event handler, you can probably keep it as is, just remove this comment